Demo/src/main/java/com/paypal/android/ui/approveorder/ApproveOrderViewModel.kt

```diff
@@ -89,47 +87,43 @@ class ApproveOrderViewModel @Inject constructor(
-                    override fun onApproveOrderSuccess(result: CardResult) {
-                        approveOrderState = ActionState.Success(result)
-                    }
-
-                    override fun onApproveOrderAuthorizationRequired(authChallenge: CardAuthChallenge) {
-                        cardClient?.presentAuthChallenge(activity, authChallenge)
-                            ?.let { presentAuthResult ->
-                                when (presentAuthResult) {
-                                    is CardPresentAuthChallengeResult.Success -> {
-                                        authState = presentAuthResult.authState
-                                    }
-
-                                    is CardPresentAuthChallengeResult.Failure -> {
-                                        approveOrderState =
-                                            ActionState.Failure(presentAuthResult.error)
-                                    }
-                                }
+                cardClient?.approveOrder(cardRequest) { result ->
+                    when (result) {
+                        is CardApproveOrderResult.Success -> {
+                            val orderInfo = result.run {
+                                OrderInfo(orderId, status, didAttemptThreeDSecureAuthentication)
                             }
-                    }
+                            approveOrderState = ActionState.Success(orderInfo)
+                        }
 
-                    override fun onApproveOrderFailure(error: PayPalSDKError) {
-                        approveOrderState = ActionState.Failure(error)
-                    }
-
-                    override fun onApproveOrderCanceled() {
-                        approveOrderState = ActionState.Failure(Exception("USER CANCELED"))
-                    }
+                        is CardApproveOrderResult.AuthorizationRequired -> {
+                            presentAuthChallenge(activity, result.authChallenge)
+                        }
 
-                    override fun onApproveOrderThreeDSecureWillLaunch() {
-                        Log.d(TAG, "3DS Auth Requested")
+                        is CardApproveOrderResult.Failure -> {
+                            approveOrderState = ActionState.Failure(result.error)
+                        }
                     }
+                }
+            }
+        }
+    }
 
-                    override fun onApproveOrderThreeDSecureDidFinish() {
-                        Log.d(TAG, "3DS Success")
-                    }
+    private fun presentAuthChallenge(
+        activity: ComponentActivity,
+        authChallenge: CardAuthChallenge
+    ) {
+        cardClient?.presentAuthChallenge(activity, authChallenge)?.let { presentAuthResult ->
+            when (presentAuthResult) {
+                is CardPresentAuthChallengeResult.Success -> {
+                    authState = presentAuthResult.authState
```

Comment:
I'm a bit lost on how this authState gets translated to success CardApproveOrderResult back to the merchant.

Reply:
So `authState` is a JSON string that allows us to continue the flow once the merchant app comes back into the foreground after browser switching. We created this pattern in BT `v5` to allow Android merchants to recover from a process kill. In BT it's called a [pendingRequest](https://github.com/braintree/braintree_android/blob/main/v5_MIGRATION_GUIDE.md#paypal).

We can always come up with a new name, `authState` for me makes sense because it's similar to the `state` parameter used in [OAuth2](https://stackoverflow.com/questions/26132066/what-is-the-purpose-of-the-state-parameter-in-oauth-authorization-request).

Reply:
I see that viewModel's completeAuthChallenge gets called in ApproveOrderView.